client-js/app/elements/labrad-registry.ts

```diff
@@ -132,6 +221,8 @@ export class LabradRegistry extends polymer.Base {
    * that contain substring in filterText
    */
   filterFunc(item) {
+    const selectedIdx = (this.$.search.value) ? 0 : this.getDefaultSelectedItem();
+    this.set('selectedIdx', selectedIdx);
```

Comment:
It seems a little bit strange to update the selected index here, especially since this gets called in a loop when the items are updated. But this might be unavoidable given the way this works now with `listItems` being a computed property. It would be nice to preserve the selected index as much as possible, for example keeping an item selected after editing it, where currently this causes the selection to be reset to the top when the item is edited and the list gets repopulated.